python/ray/runtime_context.py

```diff
@@ -405,8 +405,10 @@ def was_current_actor_reconstructed(self):
         assert (
             not self.actor_id.is_nil()
         ), "This method should't be called inside Ray tasks."
-        actor_info = ray._private.state.actors(self.actor_id.hex())
-        return actor_info and actor_info["NumRestarts"] != 0
```

Comment:
this is a core API that needs to function w/o the dashboard, so we should leave it as-is

note that it's ok for core files in `python/ray/*.py` to use private APIs